FreeRTOS/FreeRTOS-Plus-TCP
# Add backward compatibility with main branch with ipconfigIPv4_BACKWARD_COMPATIBLE

Description
-----------
This PR adds backward compatibility with main IPv4 single endpoint branch by adding following changes:
- Add FreeRTOS_GetAddressConfiguration and FreeRTOS_SetAddressConfiguration
- Rename FreeRTOS_GetUDPPayloadBuffer_ByIPType to FreeRTOS_GetUDPPayloadBuffer_Multi
- Rename xApplicationDNSQueryHook to xApplicationDNSQueryHook_Multi when ipconfigIPv4_BACKWARD_COMPATIBLE is disabled
- Rename vApplicationIPNetworkEventHook to vApplicationIPNetworkEventHook_Multi when ipconfigIPv4_BACKWARD_COMPATIBLE is disabled

Test Steps
-----------
Unit tests pass

Checklist:
----------


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
